plugin.audio.eco99music/addon.py

```diff
@@ -8,100 +8,129 @@
     channels = {}
     index = 1
 
-    for item in rss.find_all('item'):
+    for item in rss["rss"]["channel"]["item"]:
         channels.update({
             index: {
-                'album_cover': re.search("src='([^']+)'", item.find('description').string).group(1),
-                'title': item.find('title').string,
-                'description': item.find('itunes:summary').string,
-                'url': build_url({'mode': 'playlist', 'url': item.find('link').string})
+                'album_cover':
+                re.search("src='([^']+)'", item['description']).group(1),
+                'title':
+                item['title'],
+                'description':
+                item['itunes:summary'],
+                'url':
+                build_url({
+                    'mode': 'playlist',
+                    'url': item['link']
+                })
             }
         })
         index += 1
     return channels
 
 
 def get_playlists(url):
-    """
-    Extract playlists from rss
-    :return: list of playlists
+    """Get playlists of a channel.
+
+    :param url: Channel rss url.
+    :type url: str
+    :return: Dictionary containing playlist items.
+    :rtype: dict
     """
     rss = get_rss(url)
     playlists = {}
     index = 1
 
-    for item in rss.find_all('item'):
+    for item in rss["rss"]["channel"]["item"]:
         playlists.update({
             index: {
-                'album_cover': re.search("src='([^']+)'", item.find('description').string).group(1),
-                'title': item.find('title').string,
-                'description': item.find('itunes:summary').string,
-                'url': build_url({'mode': 'stream', 'url': item.find('guid').string})
+                'album_cover':
+                re.search("src='([^']+)'", item['description']).group(1),
+                'title':
+                item['title'],
+                'description':
+                item['itunes:summary'],
+                'url':
+                build_url({
+                    'mode': 'stream',
+                    'url': item['guid']
+                })
             }
         })
         index += 1
     return playlists
 
 
 def build_menu(items, is_folder):
-    """
-    Build menu control
-    :param items: items of list
-    :param is_folder: indicates whether list items are folders
-    :return:
+    """Build menu control
+
+    :param items: List of items, can be channels or playlist.
+    :type items: list
+    :param is_folder: If True the item is channel else a playlist.
+    :type is_folder: bool
     """
     items_list = []
 
     for item in items:
         # create a list item using the song filename for the label
         li = xbmcgui.ListItem(label=items[item]['title'])
         # set the fanart to the album cover
-        li.setProperty('fanart_image', os.path.join(ADDON_FOLDER, 'resources/media/fanart.jpg'))
+        li.setProperty(
```

Comment:
Please use `listitem.setArt`

Reply:
I actually see that it's done at line 132-133